src/main/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSink.java

```diff
@@ -255,4 +263,12 @@ private static void mergeInto(OutputStream out, Path directory, Configuration co
 
         return parts;
     }
+
+    private static String makeFilePathAbsolute(String path){
+        if(BucketUtils.isCloudStorageUrl(path) || BucketUtils.isHadoopUrl(path)){
+            return path;
+        } else {
+            return new File(path).getAbsolutePath();
+        }
+    }
```

Comment:
How about at least calling `mergeInto()` from a test on a directory with no bam fragments, and confirming that it throws?